optimum/exporters/openvino/model_patcher.py

```diff
@@ -2712,7 +2712,14 @@ def patched_forward(*args, **kwargs):
 
     def __enter__(self):
         super().__enter__()
-        if is_transformers_version(">=", "4.45.0"):
+
+        if is_transformers_version(">=", "4.47.0"):
+            from transformers.models.gemma2.modeling_gemma2 import GEMMA2_ATTENTION_FUNCTION
+
+            GEMMA2_ATTENTION_FUNCTION["original_eager"] = GEMMA2_ATTENTION_FUNCTION["eager"]
+            GEMMA2_ATTENTION_FUNCTION["eager"] = GEMMA2_ATTENTION_FUNCTION["sdpa"]
```

Comment:
why is this needed ? also looks like GEMMA2_ATTENTION_FUNCTION will be removed in v4.48 
https://github.com/huggingface/transformers/blob/main/src/transformers/models/gemma2/modeling_gemma2.py

Reply:
openvino attention optimizations and paged attention support va vllm requires to use sdpa, while default implementation for gemma2 is eager in transformers due to some accuracy issues with running model on cude in bf16. I think it is better to move such patching on model loading level:
https://github.com/huggingface/optimum-intel/blob/main/optimum/exporters/openvino/__main__.py#L52

for avoid necessarily to align this code refactoring sensitive patch

Reply:
> why is this needed ? also looks like GEMMA2_ATTENTION_FUNCTION will be removed in v4.48

It's the new equivalent to the lines that come afterwards, transformers seem to be migrating from a class based attention dispatching to a functional one.

>  I think it is better to move such patching on model loading level

Thanks ! will use that instead then.